pkg/controllermanager/controller/seed/seed_control.go

```diff
@@ -307,3 +329,56 @@ func (c *defaultControl) updateSeedStatus(seed *gardenv1beta1.Seed, updateCondit
 
 	return err
 }
+
+func deployBackupBucketInGarden(ctx context.Context, k8sGardenClient client.Client, seed *gardenv1beta1.Seed) error {
+	var region string
+	if seed.Spec.Backup.Region != nil {
+		region = *seed.Spec.Backup.Region
+	} else {
+		// We at this stage we assume the seed.Spec.Backup.Provider mathes the seed.Spec.Backup.Cloud as per the validation
+		// logic. And no need to validate it here again.
+		region = seed.Spec.Cloud.Region
+	}
+
+	backupBucket := &gardencorev1alpha1.BackupBucket{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: common.GenerateBackupBucketName(string(seed.Spec.Backup.Provider), region, seed.UID),
+		},
+	}
+
+	backupSecret, err := common.GetSecretFromSecretRef(ctx, k8sGardenClient, &seed.Spec.Backup.SecretRef)
+	if err != nil {
+		return err
+	}
+	checksum := common.ComputeSecretCheckSum(backupSecret.Data)
+	ownerRef := metav1.NewControllerRef(seed, gardenv1beta1.SchemeGroupVersion.WithKind("Seed"))
+
+	return kutil.CreateOrUpdate(ctx, k8sGardenClient, backupBucket, func() error {
+		backupBucket.ObjectMeta.Annotations = map[string]string{
+			common.SecretRefChecksumAnnotation: checksum,
+		}
+		backupBucket.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+		backupBucket.Spec.Provider = gardencorev1alpha1.Provider{
+			Type:   string(seed.Spec.Backup.Provider),
+			Region: region,
+		}
+		backupBucket.Spec.SecretRef = corev1.SecretReference{
+			Name:      seed.Spec.Backup.SecretRef.Name,
+			Namespace: seed.Spec.Backup.SecretRef.Namespace,
+		}
+
+		return nil
+	})
+}
+
+func validSeedAvailableForBucketRescheduling(ctx context.Context, c client.Client) bool {
+	seeds := &gardenv1beta1.SeedList{}
+	if err := c.List(ctx, seeds); err != nil {
```

Comment:
You probably meant `err == nil`.  Still, I'd prefer if you first return the error if it occurred, and only then (if no error occurred) loop over the list.